src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -909,6 +909,7 @@ std::pair<std::vector<String>, bool> ReplicatedMergeTreeSinkImpl<async_insert>::
         {
             if (e.code() == ErrorCodes::DUPLICATE_DATA_PART || e.code() == ErrorCodes::PART_IS_TEMPORARILY_LOCKED)
             {
+                tryLogCurrentException(log);
```

Comment:
We replace exception only for two specific errors - else exception is rethrown as is. If you think it's important to distinguish these errors (why?) - you can mention it in the newly generated LOGICAL_ERROR - in general I think it is a bad idea to log the same error twice - it will be confusing.